### ISSUE TITLE
Adding modules with errors to an SPA network obscures the real error message.

### ISSUE THREAD

**Issue by xchoo** [nengo/nengo#827]
Exceptions thrown within modules are not passed along to the user. Instead, the user just gets a very unhelpful message:
```
ValueError: <ModuleName> was not added as an attribute to of                     the SPA network and won't be detected
```
(Note: the extra spacing is also a bug)

I think this is what is happening:
1. Module is added to the SPA network (nengo registers the add successfully)
2. ```__setattr__``` function is called
3. exception is thrown (and module is not added to the SPA network
4. ```__exit__``` function is called (because of the exception)
5. Which can't find the recently added network in the modules list
6. ValueError is thrown.

Test example:
```
class TestModule(Module):
    def __init__(self, label="Test", seed=None, add_to_container=None):
        super(TestModule, self).__init__(label, seed, add_to_container)
        with self:
            raise Exception("EXCEPTION!")

with spa.SPA() as model:
    model.test = TestModule()

sim = nengo.Simulator(model)
```

Error message:
```
ValueError: <TestModule "Test"> was not added as an attribute to of                     the SPA network and won't be detected
```

**Comment by tcstewar** [comment]
Oops...  good catch.   

The problem is that ```SPA.__exit__``` isn't checking to see if it's in the process of handling an exception.  It's supposed to do that by checking its arguments, which are supposed to look like:

```
def __exit__(self, type, value, traceback):
```

So basically I think we just need a

```
    if value is not None:
        raise value
````

at the beginning and that should make it all work as expected....

**Comment by tcstewar** [comment]
I've done a PR with a fix for this: #828 

And I made an error in my post above.  For correct handling of exceptions in ```__exit__```, best practice is to do

```
if ex_type is not None:
    return False
````

(that way the exception traceback doesn't get mangled up)
